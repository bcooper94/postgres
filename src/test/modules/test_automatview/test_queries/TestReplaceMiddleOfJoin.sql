select employee.id, person_id, manager.id, employee_id from employee join manager on employee.id = employee_id;
--select employee.id, person_id, employee_id from employee join manager on employee.id = employee_id;
select employee.id, person_id, manager.id, employee_id from employee join manager on employee.id = employee_id;
--select employee.id, manager.id, person_id, employee_id from employee join manager on employee.id = employee_id;

select name, age, employee.id, executive.id from person join employee on person.id = person_id
 join manager on employee.id = employee_id join executive on manager.id = manager_id;

/*
 * Expected: select person.name, person.age, MatView.employee_id, executive.id
 * 	from person join MatView on person.id = MatView.employee_person_id
 * 	join executive on MatView.manager_id = executive.manager_id;
 */

/*
 * Actual (3/18/18):
 * SELECT person.name, person.age, Matview.employee_id, executive.id
 * FROM person JOIN Matview ON person.id = Matview.employee_person_id
 * JOIN executive ON Matview.manager_id = executive.manager_id
 */