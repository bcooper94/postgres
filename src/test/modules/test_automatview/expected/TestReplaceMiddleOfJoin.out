1
SELECT person.name AS person_name, person.age AS person_age, Matview.employee_id AS Matview_employee_id, executive.id AS executive_id FROM person INNER JOIN Matview ON person.id = Matview.employee_person_id INNER JOIN executive ON Matview.manager_id = executive.manager_id
